$OpenBSD: patch-src_gui_c,v 1.38 2013/11/06 07:29:30 dcoppa Exp $

Set window properties to be common for all the created windows
(upstream svn revision r2461)

Fix problem with remembered softvol not being loaded

--- src/gui.c.orig	Fri Aug 16 16:40:22 2013
+++ src/gui.c	Wed Nov  6 08:16:23 2013
@@ -1377,7 +1377,9 @@ void create_folder_progress_window()
     cancel_folder_load = FALSE;
 
     folder_progress_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
-    gtk_window_set_modal(GTK_WINDOW(folder_progress_window), TRUE);
+    gtk_window_set_type_hint(GTK_WINDOW(folder_progress_window), GDK_WINDOW_TYPE_HINT_UTILITY);
+    gtk_window_set_transient_for(GTK_WINDOW(folder_progress_window), GTK_WINDOW(window));
+    gtk_window_set_position(GTK_WINDOW(folder_progress_window), GTK_WIN_POS_CENTER_ON_PARENT);
     gtk_window_set_icon(GTK_WINDOW(folder_progress_window), pb_icon);
     gtk_window_set_resizable(GTK_WINDOW(folder_progress_window), FALSE);
     gtk_widget_set_size_request(folder_progress_window, 400, -1);
@@ -1706,11 +1708,6 @@ gboolean set_volume(void *data)
         if (!(softvol || audio_device.type == AUDIO_TYPE_SOFTVOL)) {
             gm_audio_get_volume(&audio_device);
         }
-        if (pref_volume != -1) {
-            audio_device.volume = (gdouble) pref_volume / 100.0;
-            gmtk_media_player_set_volume(GMTK_MEDIA_PLAYER(media), audio_device.volume);
-            pref_volume = -1;
-        }
         gm_log(verbose, G_LOG_LEVEL_DEBUG, "data is null new volume is %f", audio_device.volume);
 #ifdef GTK2_12_ENABLED
         gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider), audio_device.volume);
@@ -3236,6 +3233,10 @@ void menuitem_open_location_callback(GtkMenuItem * men
     GtkWidget *open_button;
 
     open_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_type_hint(GTK_WINDOW(open_window), GDK_WINDOW_TYPE_HINT_UTILITY);
+    gtk_window_set_resizable(GTK_WINDOW(open_window), FALSE);
+    gtk_window_set_transient_for(GTK_WINDOW(open_window), GTK_WINDOW(window));
+    gtk_window_set_position(GTK_WINDOW(open_window), GTK_WIN_POS_CENTER_ON_PARENT);
     gtk_window_set_icon(GTK_WINDOW(open_window), pb_icon);
 
     gtk_window_set_resizable(GTK_WINDOW(open_window), FALSE);
@@ -4827,6 +4828,10 @@ void menuitem_advanced_callback(GtkMenuItem * menuitem
     adv_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
     gtk_window_set_type_hint(GTK_WINDOW(adv_window), GDK_WINDOW_TYPE_HINT_UTILITY);
     gtk_window_set_resizable(GTK_WINDOW(adv_window), FALSE);
+    gtk_window_set_icon(GTK_WINDOW(adv_window), pb_icon);
+    gtk_window_set_transient_for(GTK_WINDOW(adv_window), GTK_WINDOW(window));
+    gtk_window_set_position(GTK_WINDOW(adv_window), GTK_WIN_POS_CENTER_ON_PARENT);
+
     gtk_window_set_title(GTK_WINDOW(adv_window), _("Video Picture Adjustments"));
 
 #if GTK_MAJOR_VERSION == 3 && GTK_MINOR_VERSION >= 2
@@ -5310,6 +5315,10 @@ void menuitem_config_callback(GtkMenuItem * menuitem, 
     GdkModifierType modifier;
 
     config_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_type_hint(GTK_WINDOW(config_window), GDK_WINDOW_TYPE_HINT_UTILITY);
+    gtk_window_set_resizable(GTK_WINDOW(config_window), FALSE);
+    gtk_window_set_transient_for(GTK_WINDOW(config_window), GTK_WINDOW(window));
+    gtk_window_set_position(GTK_WINDOW(config_window), GTK_WIN_POS_CENTER_ON_PARENT);
     gtk_window_set_icon(GTK_WINDOW(config_window), pb_icon);
 
     gtk_window_set_resizable(GTK_WINDOW(config_window), FALSE);
